src/ckeditor.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import getEditorNamespace from './utils/get-editor-namespace.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => type in { classic: 1, inline: 1 }
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.11.4/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	mounted() {
+		getEditorNamespace( this.editorUrl ).then( () => {
+			const config = { ...this.config };
+
+			// Overwrite config only if `component.readOnly` is set.
+			if ( this.readOnly !== null ) {
+				config.readOnly = this.readOnly;
+			}
+
+			const method = this.type === 'inline' ? 'inline' : 'replace';
+			const element = this.$el.firstElementChild;
+			const editor = this.instance = CKEDITOR[ method ]( element, config );
+
+			editor.on( 'instanceReady', () => {
+				const undo = editor.undoManager;
+				const data = this.value;
+
+				undo && undo.lock();
+
+				editor.setData( data, { callback: () => {
+					this.$_setUpEditorEvents();
+
+					const newData = editor.getData();
+
+					// Locking undoManager prevents 'change' event.
+					// Trigger it manually to update bound data.
+					if ( data !== newData ) {
+						this.$once( 'input', () => {
+							this.$emit( 'ready', editor );
+						} );
+
+						this.$emit( 'input', newData );
+					} else {
+						this.$emit( 'ready', editor );
+					}
+
+					undo && undo.unlock();
+				} } );
+			} );
+		} );
+	},
+
+	beforeDestroy() {
+		if ( this.instance ) {
+			this.instance.destroy();
+		}
+
+		// Note: By the time the editor is destroyed (editor.on( 'destroy' ) called)
+		// the Vue component will not be able to emit any longer. So emitting #destroy a bit earlier.
+		this.$emit( 'destroy', this.instance );
+	},
+
+	watch: {
+		value( val ) {
+			if ( this.instance.getData() !== val ) {
+				this.instance.setData( val );
+			}
+		},
+
+		readOnly( val ) {
```

Comment:
Seems like no unit test coverage here.

Reply:
Is there any particular case you'd like to cover? I've added plenty of test for `readOnly`, ones that directly check commented setter are here:
https://github.com/ckeditor/ckeditor4-vue/blob/7ba93dcf10ddba536d473795a4607d25b60abbe1/tests/component.js#L234-L244

Reply:
Removing this method from class results in the same failing tests as before, so looks like no test coverage or tests are failing positive.

Reply:
Was it already covered cc @jacekbogdanski?

Reply:
You can just remove https://github.com/ckeditor/ckeditor4-vue/blob/14409e98ac7fe09b601ca952cb11d8b775f9cba7/src/ckeditor.js#L106-L108 from code to see it yourself, but yeah, couple of tests are dying now :skull: 
![image](https://user-images.githubusercontent.com/10279586/72555610-3517e200-389d-11ea-857d-66497dbb6fab.png)